server/api/src/rsPhyBundleColl.cpp

```diff
@@ -493,8 +486,13 @@ bundleAndRegSubFiles( rsComm_t *rsComm, int l1descInx, char *phyBunDir,
         // =-=-=-=-=-=-=-
         // JMC - backport 4528
         if ( chksumFlag != 0 ) {
-            status = fileChksum( rsComm, regReplicaInp.destDataObjInfo->filePath,
-                                 subPhyPath, regReplicaInp.destDataObjInfo->rescHier, 0, tmpBunReplCache->chksumStr );
+            status = file_checksum(rsComm,
+                                   regReplicaInp.destDataObjInfo->filePath,
+                                   subPhyPath,
+                                   regReplicaInp.destDataObjInfo->rescHier,
+                                   nullptr,
+                                   regReplicaInp.destDataObjInfo->dataSize, // TODO Should this come from the source data object info?
```

Comment:
where is it coming from now?   isn't `destDataObjInfo->dataSize` populated with the size from the source already?   or is it coming from a physical stat?  seems too early for that...

Reply:
I'm not sure where this comes from. The CI tests that hit this code appear to behave so this may be fine. I'll spend a few minutes looking over this.

Reply:
Reading the code, the data size member of the src/dst dataObjInfo appears to be zero at the point of the `file_checksum` invocation. I don't see the data size member being updated anywhere. The `regReplicaInp` is constructed and initialized within this function.

Not only that, but the phybun APIs and commands appear to be marked as deprecated (see https://github.com/irods/irods/issues/3603).

The tests that invoke `iphybun` are skipped or commented out. [This test](https://github.com/irods/irods/blob/be48c4ab2880b1188ddcf12ec2f44f8d178b434d/scripts/irods/test/test_all_rules.py#L456) is the only one that invokes a phybun operation. However, no checksum flags are passed during this test. The rule file can be found [here](https://github.com/irods/irods_client_icommands/blob/4-2-stable/test/rules/rulemsiPhyBundleColl.r).

Q. Is this API still supported?
Q. What do we want to do about this API?

Reply:
After discussing this topic with the team, it has been decided that the behavior of the phybun APIs is not something that this PR should be concerned about.